src/typing/type_inference_js.ml

```diff
@@ -1292,8 +1292,21 @@ and toplevels cx stmts =
       if uc < List.length stmts
       then (
         let msg = "unreachable code" in
-        let (loc, _) = List.nth stmts uc in
-        Flow_js.add_warning cx [mk_reason "" loc, msg]
+        let rec drop n lst =
+          match (n, lst) with
+          | (_, []) -> []
+          | (0, l) -> l
+          | (x, h :: t) -> drop (x-1) t
+        in
+        let trailing = drop uc stmts in
+        trailing |> List.iter Ast.Statement.(fun stmt ->
```

Comment:
:+1: 

While we're at telling "good" OCaml guideline, in your third pattern branch in `drop`, you don't use `h` so you can shadow it by replacing it with `_` like `(x, _ :: t)`.

Also most OCamllers love to use `pred x` instead of `x - 1` (same thing with `succ x` and `x + 1`).